contracts/BrickFarming.sol

```diff
@@ -0,0 +1,297 @@
+
+        emit Deposit(msg.sender, _pid, _amount);
+    }
+
+    function withdraw(uint256 _pid, uint256 _amount)
+        external
+        nonReentrant
+        whenNotPaused
+        validatePoolByPid(_pid)
+    {
+        PoolInfo storage pool = poolInfo[_pid];
+        UserInfo storage user = userInfo[_pid][msg.sender];
+
+        require(user.amount >= _amount, "BrickFarm: Invalid withdraw amount");
+
+        uint256 feeAmount;
+        if (block.timestamp < user.lastClaim.add(pool.lockupDuration)) {
+            feeAmount = _amount.mul(emergencyWithdrawFee).div(FEE_MULTIPLIER);
+        }
+
+        uint256 pendingRewards = pendingReward(_pid, msg.sender);
+        if (pendingRewards > 0) {
+            uint256 sentRewards = safeTokenTransfer(msg.sender, pendingRewards);
+            emit Claim(msg.sender, _pid, sentRewards);
+            user.pendingRewards = pendingRewards.sub(sentRewards);
+        }
+
+        if (_amount > 0) {
+            user.amount = user.amount.sub(_amount);
+            user.lastClaim = block.timestamp;
+
+            pool.stakedToken.safeTransfer(
+                address(msg.sender),
+                _amount.sub(feeAmount)
+            );
+
+            if (feeAmount > 0) {
+                pool.stakedToken.safeTransfer(treasury, feeAmount);
+            }
+        }
+
+        user.rewardDebt = user.amount.mul(pool.accTokenPerShare).div(
+            SHARE_MULTIPLIER
+        );
+        pool.totalAmount = pool.totalAmount.sub(_amount);
+
+        emit Withdraw(msg.sender, _pid, _amount);
+    }
+
+    function safeTokenTransfer(address _to, uint256 _amount)
+        internal
+        returns (uint256)
+    {
+        uint256 rewardTokenBal = rewardToken.balanceOf(address(this));
+        if (_amount > rewardTokenBal) {
+            rewardToken.safeTransfer(_to, rewardTokenBal);
+            return rewardTokenBal;
+        } else {
+            rewardToken.safeTransfer(_to, _amount);
+            return _amount;
+        }
+    }
+
+    function depositReward(uint256 _amount) external {
+        require(_amount > 0, "BrickFarm: Invalid amount");
+
+        rewardToken.safeTransferFrom(msg.sender, address(this), _amount);
+
+        uint256 length = poolInfo.length;
+        for (uint256 pid = 0; pid < length; ++pid) {
+            PoolInfo storage pool = poolInfo[pid];
+
+            pool.accTokenPerShare = pool.accTokenPerShare.add(
+                _amount.mul(pool.allocPoint).mul(SHARE_MULTIPLIER).div(
+                    totalAllocPoint.mul(pool.totalAmount)
+                )
+            );
+        }
+    }
+
+    function add(
+        uint256 _allocPoint,
+        IERC20 _stakedToken,
+        uint256 _lockupDuration
+    ) external onlyOwner {
+        require(
+            !isPoolAdded[address(_stakedToken)],
+            "BrickFarm: There's already a pool with that token!"
+        );
+        require(_lockupDuration > 0, "BrickFarm: Invalid lockupDuration");
```

Comment:
@bejaxer what if we decide not to add lockup duration?

Reply:
> @bejaxer what if we decide not to add lockup duration?

Then we can remove this require

Reply:
yes let's remove it